ledger/internal/cow.go

```diff
@@ -38,7 +38,12 @@ import (
 //                  ||     ||
 
 type roundCowParent interface {
-	lookup(basics.Address) (basics.AccountData, error)
+	lookup(basics.Address) (ledgercore.AccountData, error)
+	lookupAppParams(addr basics.Address, aidx basics.AppIndex, fromCache bool) (ledgercore.AppParamsDelta, bool, error)
+	lookupAssetParams(addr basics.Address, aidx basics.AssetIndex, fromCache bool) (ledgercore.AssetParamsDelta, bool, error)
+	lookupAppLocalState(addr basics.Address, aidx basics.AppIndex, fromCache bool) (ledgercore.AppLocalStateDelta, bool, error)
+	lookupAssetHolding(addr basics.Address, aidx basics.AssetIndex, fromCache bool) (ledgercore.AssetHoldingDelta, bool, error)
+
```

Comment:
Two things about `fromCache`. 1) `cacheOnly` would be a clearer name, since I might imagine that fromCache=false means "Do no look in cache". But it seems to mean that it's ok to look beyond the cache. 2) This would be a good place for a comment explaining the intuition for why you might use this flag.  I have one idea, but it's only a guess.

I'm not sure my idea is right, because I now realize this means "Don't look in my parent". But the parent of a roundCow might be another roundCowState, which is in memory and fast, right?  It doesn't mean what I thought it meant - don't go to disk.

Reply:
updated with:
```
    // lookup retrieves data about an address, eventually querying the ledger if the address was not found in cache.
    lookup(basics.Address) (ledgercore.AccountData, error)

    // lookupAppParams, lookupAssetParams, lookupAppLocalState, and lookupAssetHolding retrieve data for a given address and ID.
    // If cacheOnly is set, the ledger DB will not be queried, and only the cache will be consulted.
    // This is used when we know a given value is already in cache (from a previous query for that same address and ID),
    // and would rather have an error returned if that assumption is wrong, rather than hit the ledger.
    lookupAppParams(addr basics.Address, aidx basics.AppIndex, cacheOnly bool) (ledgercore.AppParamsDelta, bool, error)
    lookupAssetParams(addr basics.Address, aidx basics.AssetIndex, cacheOnly bool) (ledgercore.AssetParamsDelta, bool, error)
    lookupAppLocalState(addr basics.Address, aidx basics.AppIndex, cacheOnly bool) (ledgercore.AppLocalStateDelta, bool, error)
    lookupAssetHolding(addr basics.Address, aidx basics.AssetIndex, cacheOnly bool) (ledgercore.AssetHoldingDelta, bool, error)
```